dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java

```diff
@@ -1150,7 +1153,8 @@ private void initComplementDataParam(ProcessDefinition processDefinition,
             complementDate = CronUtils.getSelfScheduleDateList(cmdParam);
         }
 
-        if (complementDate.size() > 0 && Flag.NO == processInstance.getIsSubProcess()) {
+        if (complementDate != null && complementDate.size() > 0 && Flag.NO == processInstance
```

Comment:
I am used to use ``` CollectionUtils.isEmpty(complementDate) && Flag.NO == processInstance ``` to make this judgment. 

it looks like it will make the code more concise ，The code logic of this method is as follows：
```
    public static boolean isEmpty(Collection coll) {
        return coll == null || coll.isEmpty();
    }
```

Reply:
I made this suggestion because I saw that you mentioned code clean

Reply:
> I made this suggestion because I saw that you mentioned code clean

Sure, u are correct, i will update it.